fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Field.java

```diff
@@ -169,11 +168,16 @@ public List<Integer> getTmpUsedStructFieldPos() {
         return tmpUsedStructFieldPos;
     }
 
-    public boolean matchesPrefix(TableName prefix) {
-        if (relationAlias != null) {
-            return relationAlias.getTbl().equals(prefix.getTbl());
+    public boolean matchesPrefix(TableName tableName) {
+        if (tableName.getCatalog() != null && !tableName.getCatalog().equals(relationAlias.getCatalog())) {
```

Comment:
Can  relationAlias be null?

Reply:
relationAlias can not be null

    /**
     * TableName of field
     * relationAlias is origin table which table name is explicit, such as t0.a
     * Field come from scope is resolved by scope relation alias,
     * such as subquery alias and table relation name
     */